ops/testing.py

```diff
@@ -746,7 +815,9 @@ def get_relation_data(self, relation_id: int, app_or_unit: AppUnitOrName) -> typ
         """
         if hasattr(app_or_unit, 'name'):
             app_or_unit = app_or_unit.name
-        return self._backend._relation_data[relation_id].get(app_or_unit, None)
+        with self.event_context(''):
+            # disable strict access control
+            return self._backend._relation_data[relation_id].get(app_or_unit, None)
```

Comment:
It would be unnecessary if the user were running that call from test code, normally.
However, suppose that the user calls `harness.get_relation_data` while inside a debugging session (e.g. put a breakpoint in `Charm._on_pebble_ready`), so while a hook IS in fact running. Then if the user were to do `harness.set_relation_data`, it would fail because you're inside a hook context).
My reasoning was: we want to ensure that `harness.set_relation_data` ALWAYS works the same, i.e. bypasses all checks.
Hence the `event_context('')`: this tells the harness: do this call always as if no hook were running.

Reply:
It becomes redundant now that I separate the raw data in the backend and use RelationDataContent to access it.